apps/betterangels-backend/shelters/widgets.py

```diff
@@ -39,3 +42,130 @@ def compress(self, data_list: List[float]) -> Optional[str]:
             point_str = "POINT(%f %f)" % tuple(reversed(data_list))
             return point_str
         return None
+
+
+class TimeRangeWidget(forms.TextInput):
+    template_name: str = "shelters/time_range_widget.html"
+
+    def get_context(self, name: str, value: Any, attrs: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
+        # Convert the value into a display string if it's a list.
+        if value is None:
+            display_value = ""
+        elif isinstance(value, list):
+            display_value = ", ".join(f"{start.strftime('%I:%M%p')}-{end.strftime('%I:%M%p')}" for start, end in value)
+        else:
+            display_value = value
+
+        # Call the parent to build the default context.
+        context = super().get_context(name, display_value, attrs)
+        context["widget"]["flat_attrs"] = flatatt(context["widget"]["attrs"])
+        # Optionally, add any extra context needed for your custom HTML.
+        return context
+
+
+class TimeRangeFormField(forms.Field):
+    widget = TimeRangeWidget
+
+    def to_python(self, value: Any) -> Optional[List[Tuple[datetime.time, datetime.time]]]:
+        if not value:
+            return []
+        if isinstance(value, list):
+            return value
+        if isinstance(value, str):
+            try:
+                ranges: List[Tuple[datetime.time, datetime.time]] = []
+                # Expect input like: "8:00AM-12:00PM, 1:00PM-5:00PM, 6:00PM-10:00PM"
+                for rng in value.split(","):
+                    rng = rng.strip()
+                    if not rng:
+                        continue
+                    start_str, end_str = rng.split("-")
+                    start_time = datetime.datetime.strptime(start_str.strip(), "%I:%M%p").time()
+                    end_time = datetime.datetime.strptime(end_str.strip(), "%I:%M%p").time()
+                    if start_time >= end_time:
+                        raise forms.ValidationError("Each time range must have a start time before the end time.")
+                    ranges.append((start_time, end_time))
+                return ranges
+            except Exception as e:
+                raise forms.ValidationError(
+                    "Invalid format for time ranges. Expected format: '8:00AM-12:00PM, 1:00PM-5:00PM, ...'"
+                ) from e
+        raise forms.ValidationError("Invalid input type for MultiTimeRangeFormField")
+
+    def validate(self, value: Any) -> None:
+        super().validate(value)
+        for start, end in value:
+            if start >= end:
+                raise forms.ValidationError("Each time range must have a start time before the end time.")
+
+
+class TimeRangeField(models.Field):
```

Comment:
If we use a JSONField this will prevent us from using the built in Postgres' built in date/time features. I.e being able to efficiently look up times, overlapping date ranges, etc.